GitUI/Avatars/CacheControlAvatarProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+
+namespace GitUI.Avatars
+{
+    /// <summary>
+    /// A helper provider that wraps an <see cref="IAvatarProvider"/>
+    /// and manages the cache of related <see cref="ICachedAvatarProvider"/>s.
+    /// </summary>
+    public sealed class CacheControlAvatarProvider : ICachedAvatarProvider
+    {
+        private readonly IAvatarProvider _rootProvider;
+        private readonly ICachedAvatarProvider[] _cachedProviders;
+
+        public CacheControlAvatarProvider(
+            [NotNull] IAvatarProvider rootProvider,
+            params ICachedAvatarProvider[] cachedProviders)
+        {
+            _rootProvider = rootProvider ?? throw new ArgumentNullException(nameof(rootProvider));
+            _cachedProviders = cachedProviders ?? throw new ArgumentNullException(nameof(cachedProviders));
```

Comment:
It can be null. There is also a test for that in `CacheControlAvatarProviderTests` called `Construction_with_null_parameters_is_permitted` (the second assert/case).

Reply:
I stand corrected then :) Thank you for checking.